.github/workflows/docker-build.yml

```diff
@@ -1,38 +1,56 @@
-name: Docker CI workflow 
+﻿name: Docker CI workflow
 
 on:
   push:
     branches: ["main"]
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
   pull_request:
     branches: ["main"]
+    types:
+      - opened
+      - reopened
+      - synchronize
 
 jobs:
-
   docker-build:
     runs-on: ubuntu-latest
-    name: Run docker build  
+    name: Run docker build
     steps:
-      
       - uses: actions/checkout@v4
 
       - name: Docker Login
```

Comment:
Are these 2 redundant? Why we need to login 2 times.

Reply:
I need this login to push images to github registry